### ISSUE TITLE
Confusion in metric.py

### ISSUE THREAD

**Issue by Shengqi77** [EPFL-VILAB/ViPer#9]
What do the numbers 648 and 387 represent? Do they stand for non-preferred and preferred? And why specifically 648 and 387?    score = torch.exp(logits[:, -1][:, 648]) / (torch.exp(logits[:, -1][:, 648]) + torch.exp(logits[:, -1][:, 387]))

**Comment by sogandstormesalehi** [comment]
Yes, they represent preferred (648) and non-preferred (387). These are token IDs for + and - .You can verify this with tokenizer.convert_ids_to_tokens(648)

**Comment by Shengqi77** [comment]
Thanks！




发件人："Storme" ***@***.***>
发送日期：2025-04-03 19:01:29
收件人："EPFL-VILAB/ViPer" ***@***.***>
抄送人：
主 题：Re: [EPFL-VILAB/ViPer] Confusion in metric.py (Issue #9)


Yes, they represent preferred (648) and non-preferred (387). These are token IDs for + and - .You can verify this with tokenizer.convert_ids_to_tokens(648)

—
Reply to this email directly, view it on GitHub, or unsubscribe.
You are receiving this because you authored the thread.Message ID: ***@***.***>

sogandstormesalehi left a comment (EPFL-VILAB/ViPer#9)

Yes, they represent preferred (648) and non-preferred (387). These are token IDs for + and - .You can verify this with tokenizer.convert_ids_to_tokens(648)
